py/objstr.c

```diff
@@ -1912,10 +1912,7 @@ mp_int_t mp_obj_str_get_buffer(mp_obj_t self_in, mp_buffer_info_t *bufinfo, mp_u
         bufinfo->typecode = 'B'; // bytes should be unsigned, so should unicode byte-access
         return 0;
     } else {
-        // can't write to a string
-        bufinfo->buf = NULL;
-        bufinfo->len = 0;
-        bufinfo->typecode = -1;
+        // can't write to a string.
```

Comment:
Minor thing: please don't update comments if they don't need it, like this.  (Yes comments are a bit inconsistent in their formatting, but that's a separate issue :)